libp2p/peer/id.py

```diff
@@ -89,7 +89,7 @@ def from_pubkey(cls, key: PublicKey) -> "ID":
         return cls(mh_digest.encode())
 
 
-def digest(data: Union[str, bytes]) -> bytes:
+def sha256_digest(data: Union[str, bytes]) -> bytes:
```

Comment:
i don't think we need to block merging for this but something to consider is the name of this function...

making it `sha256_digest` makes it very specific to producing a SHA-256 hash, which is what we want to do here. however, consider what happens in the future if the kademlia code changes again to use a different hash -- we will need to go and find all of the uses of `sha256_digest` and update them. also in terms of communicating information, the name `sha256_digest` doesn't really add more information than what is already provided by `hashlib.sha256`.

in general, i aim to name things based on what they provide, not what they do. this generally means better maintenance over time and hopefully better communicates intent to myself and others down the line. in this case, i'd think to name it something like `dht_digest` and also probably move it to the (future) `libp2p.kademlia` module. my two cents on naming :)